### PR TITLE
FIX - Quest - Exorcism of Colonel Jules

### PR DESCRIPTION
**Pull request by danlapps**
Thanks to WoW-Studio@ 0aa81ca for doing most of the work

Trying to convert

Thank you @Rushor for doing most of the converting work :)
@Nyeriah for pointing all the errors
@tkrokli for the squasing tutorial

### CODE REVIEW

**Comment by danlapps** [comment]
Might be missing some gossip menu options will have to check

**Comment by DarkBrain2580** [comment]
Error in:
UPDATE `creature_template` SET `ScriptName`= 'npc_colonel_jules' WHERE `entry`= 22431;

It must be:
UPDATE `creature_template` SET `ScriptName`= 'npc_colonel_jules' WHERE `entry`= 22432;

**Comment by danlapps** [comment]
Tested and works.  Some details to adjust : 
```checklist
Darkness Released seems to be invisible
Banara runs to the bed instead of waiting a bit then walking.
Talking to Jules before triggering the quest should trigger some "possessed" text instead of the "hello how are you doing" text.
```

**Comment by Aokromes** [comment]
Failed — The Travis CI build failed

File: sql/updates/world/2015_03_19_01_world.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
+UPDATE `creature_template` SET `ScriptName`= 'npc_barada' WHERE `entry`= 22431;
+UPDATE `creature_template` SET `ScriptName`= 'npc_colonel_jules' WHERE `entry`= 22432;
+
+DELETE FROM `gossip_menu_option` WHERE `menu_id` = 8539;
+INSERT INTO `gossip_menu_option` (`menu_id`, id, option_icon, `option_text`, `OptionBroadcastTextID`, `option_id`, `npc_option_npcflag`) VALUES
```
Comment (Nyeriah) [inline sql/updates/world/2015_03_19_01_world.sql]: missing some ` ` here

**Comment by Nyeriah** [comment]
The command "mysql -utrinity -ptrinity world < sql/base/dev/world_database.sql" exited with 0.
0.54s$ cat sql/updates/world/*.sql | mysql -utrinity -ptrinity world
ERROR 1064 (42000) at line 17771: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' AND `link`=3' at line 1

**Comment by Nyeriah** [comment]
> Failed — The Travis CI build failed

https://github.com/TrinityCore/TrinityCore/commit/ffa5ef1a2e23e8b8fdb681870dc7a125b1ccdfff it's cause of this :P

**Comment by danlapps** [comment]
@maxxx it's been updated in last commit, not sure if you are checking wrong one

**Comment by DarkBrain2580** [comment]
@danlapps you thanks everyone but not the Team from WoW Studio. So thats not ok.

**Comment by danlapps** [comment]
Indeed my bad, correcting

**Comment by DarkBrain2580** [comment]
thx^^

File: src/server/scripts/Outland/zone_hellfire_peninsula.cpp [hunk 2]
```diff
@@ -420,10 +420,462 @@ class npc_fel_guard_hound : public CreatureScript
...
+    SAY_JULES_3  = 2,
+    SAY_JULES_4  = 3,
+    SAY_JULES_5  = 4,
+};
+
+Position const exorcismPos[11] =
+{
+    { -707.123f, 2751.686f, 101.592f, 4.577416f }, //Barada Waypoint-1      0
+    { -710.731f, 2749.075f, 101.592f, 1.513286f }, //Barada Cast position   1
+    { -710.332f, 2754.394f, 102.948f, 3.207566f }, //Jules 					2
```
Comment (jackpoz) [inline src/server/scripts/Outland/zone_hellfire_peninsula.cpp]: tabs

File: src/server/scripts/Outland/zone_hellfire_peninsula.cpp [hunk 3]
```diff
@@ -420,10 +420,462 @@ class npc_fel_guard_hound : public CreatureScript
...
+                                break;
+                        }
+                        break;
+                }
+            }
+        }
+
+        private:
+        EventMap events;
+        uint8 step;
```
Comment (jackpoz) [inline src/server/scripts/Outland/zone_hellfire_peninsula.cpp]: please initialize this field and "jules" in the constructor

File: src/server/scripts/Outland/zone_hellfire_peninsula.cpp [hunk 4]
```diff
@@ -420,10 +420,462 @@ class npc_fel_guard_hound : public CreatureScript
...
+                        break;
+                }
+            }
+        }
+        
+        private:
+        EventMap events;
+        SummonList summons;
+
+        uint8 circleRounds;
```
Comment (jackpoz) [inline src/server/scripts/Outland/zone_hellfire_peninsula.cpp]: please initialize these fields in the constructor

File: src/server/scripts/Outland/zone_hellfire_peninsula.cpp [hunk 5]
```diff
@@ -420,10 +420,462 @@ class npc_fel_guard_hound : public CreatureScript
...
+    struct npc_colonel_julesAI : public ScriptedAI
+    {
+        npc_colonel_julesAI(Creature* creature) : ScriptedAI(creature), summons(me) { }
+       
+        void Reset() override
+        {
+            events.Reset();
+
+            summons.DespawnAll();
+            circleRounds = 0;
```
Comment (danlapps) [inline src/server/scripts/Outland/zone_hellfire_peninsula.cpp]: @jackpoz  Isnt it the initialization you want ?

Comment (Nyeriah) [inline src/server/scripts/Outland/zone_hellfire_peninsula.cpp]: https://github.com/TrinityCore/TrinityCore/pull/14411/files#r26816432

File: src/server/scripts/Outland/zone_hellfire_peninsula.cpp [hunk 6]
```diff
@@ -420,10 +420,481 @@ class npc_fel_guard_hound : public CreatureScript
...
+                        }
+                        break;
+                }
+            }
+        }
+
+        private:
+        EventMap events;
+        uint8 step;
+        Creature* jules;
```
Comment (ccrs) [inline src/server/scripts/Outland/zone_hellfire_peninsula.cpp]: never store a pointer, get its objectguid

**Comment by Carbenium** [comment]
@danlapps: Did you deal with this in the last time?

**Comment by tkrokli** [comment]
@danlapps : would you want someone else to finish this for you? It would be easier to see the changes if I add your commits to my source and post a squashed version (I could PR it to your repo, if you like). :smirk_cat:

**Pull request closed without merging**